src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,144 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+
+    private T[] storage;
+
+    private int currentCapacity;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.storage = (T[]) new Object[DEFAULT_CAPACITY];
+        this.currentCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
+        if (this.size == this.currentCapacity) {
+            this.resizeStorage();
```

Comment:
please do not use the keyword `this` when calling methods of this class, in fact even with fields it is optional. Its purpose is to identify the variable of the current class from any external class variable with the same name (there is nothing similar in your class). It is also customary to use it in the constructor when initializing class fields